src/unistd.rs

```diff
@@ -1124,12 +1125,24 @@ pub fn getgroups() -> Result<Vec<Gid>> {
 /// specific user and group. For example, given the user `www-data` with UID
 /// `33` and the group `backup` with the GID `34`, one could switch the user as
 /// follows:
-/// ```
+///
+/// ```rust,no_run
```

Comment:
Why are you adding `no_run` ?  If this was an unreliable doc test, then you should fix it in a separate commit than the commit that fixes the doc warnings.

Reply:
It was never run before, that's the biggest part of this change. There were doctests that weren't being found and executed, so this doctest has never been run. And turns out it fails when run.

Reply:
I can put both of these changes in a separate commit that explains all this.

Reply:
Do you understand why they weren't being found?  Because I don't.

Reply:
The previous parser didn't find them, so they weren't run. I was always curious why, but I never cared enough to actually read up on it. If you ran our tests right now with the latest Rust, it'd explain this in the compiler warnings, though not give too many specifics.